utils/mcp_client.py

```
@@ -0,0 +1,668 @@
+    "password",
+    "secret",
+    "session",
+    "token",
+)
+aiohttp: Any | None = None
+
+
+class _PinnedResolver:
+    """aiohttp resolver that only returns pre-validated addresses for one host."""
+
+    def __init__(self, addresses_by_host: dict[str, list[str]]):
```

Ruff (0.15.21)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)